pkg/testframework/registry.go

```diff
@@ -482,3 +492,39 @@ func CreateEphemeralRegistry(t *testing.T, restConfig *rest.Config, namespace st
 	t.Logf("created ephemeral registry: %s (%s)", host, name)
 	return host, name, cleanup
 }
+
+func checkRoute(host string) error {
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	url := fmt.Sprintf("https://%s/v2/", host)
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == http.StatusServiceUnavailable {
```

Comment:
Do we care if the status code is something else, like 404?

In reviewing what the networking team does, their script only exits 0 if the OK (200) response is returned.

Reply:
Yes, you are right. Adjusted accordingly.

Reply:
Just clarifying this, for some tests the registry is deployed with basic auth in place. In such scenario a 403 return is also OK. I have adjusted the deployment to consider 200 or 403 as "the registry is good to go".